public/apps/configuration/panels/role-edit/tenant-panel.tsx

```diff
@@ -13,21 +13,123 @@
  *   permissions and limitations under the License.
  */
 
-import React from 'react';
-import { EuiFieldText } from '@elastic/eui';
-import { PanelWithHeader } from '../../utils/panel-with-header';
+import { EuiButton, EuiComboBox, EuiFlexGroup, EuiFlexItem, EuiSuperSelect } from '@elastic/eui';
+import React, { Dispatch, Fragment, SetStateAction } from 'react';
+import { RoleTenantPermission } from '../../types';
+import {
+  appendElementToArray,
+  removeElementFromArray,
+  updateElementInArrayHandler,
+} from '../../utils/array-state-utils';
+import { appendOptionToComboBoxHandler, stringToComboBoxOption } from '../../utils/combo-box-utils';
 import { FormRow } from '../../utils/form-row';
+import { PanelWithHeader } from '../../utils/panel-with-header';
+import { ComboBoxOptions, RoleTenantPermissionStateClass, TenantPermissionType } from './types';
+
+const TENANT_READ_PERMISSION = 'kibana_all_read';
+const TENANT_WRITE_PERMISSION = 'kibana_all_write';
+
+export function buildTenantPermissionState(
+  permissions: RoleTenantPermission[]
+): RoleTenantPermissionStateClass[] {
+  return permissions.map(permission => {
```

Comment:
Minor: It might be a bit confusion as here "permission" actually means a tenant or tenantPermission instead of read/write. Probably consider a better name for ease of understanding.

Reply:
Renamed to tenantPermission